doc/lsst.pipe.base/creating-a-task.rst

```diff
@@ -0,0 +1,350 @@
+
+- :ref:`__init__ <creating-a-task-class-init-method>`: construct and initialize a task.
+- :ref:`run <creating-a-task-class-run-method>`: process one item of data.
+
+These methods are described in more depth below.
+
+.. _creating-a-task-class-init-method:
+
+The ``__init__`` method
+-----------------------
+
+Use the ``__init__`` method (task constructor) to do the following:
+
+- Call the parent task's ``__init__`` method
+- Make subtasks by calling ``self.makeSubtask(name)``, where ``name`` is the name of a field of type `lsst.pex.config.ConfigurableField` in your :ref:`task's configuration <creating-a-task-configuration>`.
+- Make a schema if your task uses an `lsst.afw.table`.
+  For an example of such a task `lsst.pipe.tasks.calibrate.CalibrateTask`.
+- Initialize any other instance variables your task needs.
+
+Here is ``exampleCmdLineTask.ExampleCmdLineTask.__init__``:
+
+.. code-block:: python
+
+   def __init__(self, *args, **kwargs):
+       """Construct an ExampleCmdLineTask
+       Call the parent class constructor and make the "stats" subtask from the config field of the same name.
+       """
+       pipeBase.CmdLineTask.__init__(self, *args, **kwargs)
+       self.makeSubtask("stats")
+
+That task creates a subtask named ``stats`` to compute image statistics.
+Here is the ``__init__`` method for the default version of the `stats` subtask:
+
+exampleTask.ExampleSigmaClippedStatsTask, which is slightly more interesting:
+
+.. code-block:: python
+
+   def __init__(self, *args, **kwargs):
+       """!Construct an ExampleSigmaClippedStatsTask
+       The init method may compute anything that that does not require data.
+       In this case we create a statistics control object using the config
+       (which cannot change once the task is created).
+       """
+       pipeBase.Task.__init__(self, *args, **kwargs)
+       self._badPixelMask = MaskU.getPlaneBitMask(self.config.badMaskPlanes)
+       self._statsControl = afwMath.StatisticsControl()
+       self._statsControl.setNumSigmaClip(self.config.numSigmaClip)
+       self._statsControl.setNumIter(self.config.numIter)
+       self._statsControl.setAndMask(self._badPixelMask)
+
+This creates a binary mask identifying bad pixels in the mask plane and an `lsst.afw.math.StatisticsControl`, specifying how statistics are computed.
+Both of these are constants, and thus are the same for each invocation of the ``run`` method; this is strongly recommended, as explained in the next section.
+
+.. _creating-a-task-class-run-method:
+
+The run method
+--------------
+
+Most tasks have a ``run`` method which perform the task's data processing operation.
+This is required for command-line tasks and strongly recommended for most other tasks.
+One exception is if your task needs different methods to handle different data types (C++ handles this using overloaded functions, but the standard technique is Python is to provide different methods for different call signatures).
```

Comment:
I think that we decided that this wasn't acceptable except under very special circumstances.  All tasks must have a run method, and it must take a list of variables that it uses (by name, not as a dict or other blob such as a dataRef).

While this rule has not been applied much for cmdLineTasks it has been almost universally obeyed for Tasks